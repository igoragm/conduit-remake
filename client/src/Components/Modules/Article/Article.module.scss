.articlePreview {
    width: 100%;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    padding: 1.5rem 0;
}

.pullXsRight {
    float: right !important;
}

.date {
    color: #bbb;
    font-size: 0.8rem;
    display: block;
}

.author {
    display: block;
    font-weight: 500 !important;
}

.info {
    margin: 0 1.5rem 0 0.3rem;
    display: inline-block;
    vertical-align: middle;
    line-height: 1rem;
}

.articlePreviewH1 {
    font-weight: 600 !important;
    font-size: 1.5rem !important;
    margin-bottom: 3px;
}

.previewLink {
    color: inherit !important;
}

.articleMeta {
    margin: 0 0 1rem;
    display: block;
    position: relative;
    font-weight: 300;
}

.articleMetaImg {
    display: inline-block;
    vertical-align: middle;
    height: 32px;
    width: 32px;
    border-radius: 30px;
}

.articlePreview {
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    padding: 1.5rem 0;
    width: 100%;
}

.ionHeart {
    font-family: "Ionicons";
    font-style: normal;
    -webkit-font-smoothing: antialiased;

    &:before {
        content: "\f141";
    }
}

.tagList {
    padding-left: 0 !important;
    display: inline-block;
    list-style: none !important;
    float: right;
}

.tag {
    border: 1px solid #ddd;
    color: #aaa !important;
    background: 0 0 !important;
    font-size: 0.8rem;
    padding-top: 0.1rem;
    padding-bottom: 0.1rem;
    white-space: nowrap;
    margin-right: 3px;
    margin-bottom: 0.2rem;
    display: inline-block;
    padding-right: 0.6em;
    padding-left: 0.6em;
    border-radius: 10rem;
}

.tagBtn {
    border: 1px solid transparent;
    font: inherit;
    padding: 0.25rem 0.5rem;
    font-size: 0.875rem;
    border-radius: 0.2rem;
    color: #5cb85c;
    background-image: none;
    background-color: transparent;
    border-color: #5cb85c;
    display: inline-block;
    font-weight: 400;
    line-height: 1.25;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    cursor: pointer;
    user-select: none;
}
