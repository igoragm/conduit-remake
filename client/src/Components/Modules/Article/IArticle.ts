import IArticleListItem from "../ArticleList/IArticleListItem";

interface IArticle {
    favorite: (slug: string) => void;
    unfavorite: (slug: string) => void;
    article: IArticleListItem;
}

export default IArticle;
