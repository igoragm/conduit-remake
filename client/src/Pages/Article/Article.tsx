import React from "react";

export const Article = (props: any) => {
    return <div>Article page</div>;
};

export default Article;
