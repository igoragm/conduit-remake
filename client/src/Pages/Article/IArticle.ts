// tslint:disable-next-line:no-empty-interface
interface IArticle {}

export default IArticle;
