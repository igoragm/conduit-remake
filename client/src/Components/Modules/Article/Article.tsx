import React, { FC, MouseEvent } from "react";
import { Link } from "react-router-dom";
import IArticle from "./IArticle";
import { SetState } from "../../../Utils/StateHandler";
import styles from "./Article.module.scss";

const Article: FC<IArticle> = (props: IArticle) => {
    const { favorite, unfavorite, article } = props;

    const handleClick = (ev: MouseEvent<HTMLElement>) => {
        ev.preventDefault();
        if (article.favorited) {
            unfavorite(article.slug);
        } else {
            favorite(article.slug);
        }
    };

    return (
        <div data-testid="rtl-article-id" id="article-item" className={styles.articlePreview}>
            <div className={styles.articleMeta}>
                <Link to={`/@${article.author.username}`}>
                    <img
                        className={styles.articleMetaImg}
                        src={article.author.image ? article.author.image : "https://loremflickr.com/320/240"}
                        alt={article.author.username}
                    />
                </Link>

                <div className={styles.info}>
                    <Link id="author" className={styles.author} to={`/@${article.author.username}`}>
                        {article.author.username}
                    </Link>
                    <span id="createdAt" className={styles.date}>
                        {new Date(article.createdAt).toDateString()}
                    </span>
                </div>

                <div className={styles.pullXsRight}>
                    <button id="favoritesCount" className={styles.tagBtn} onClick={handleClick}>
                        <i className={styles.ionHeart} /> {article.favoritesCount}
                    </button>
                </div>
            </div>

            <Link id="article-body" target="_blank" to={`/article/${article.slug}`} className={styles.previewLink}>
                <h1 className={styles.articlePreviewH1} id="articleTitle">
                    {article.title}
                </h1>
                <p id="articleDescription">{article.description}</p>
                <span>Read more...</span>
                <ul id="tags-list" className={styles.tagList}>
                    {article.tagList.map((tag: string) => {
                        const id = `articleTag_${tag}`;
                        return (
                            <li id={id} className={styles.tag} key={tag}>
                                {tag}
                            </li>
                        );
                    })}
                </ul>
            </Link>
        </div>
    );
};

export default SetState(Article);
