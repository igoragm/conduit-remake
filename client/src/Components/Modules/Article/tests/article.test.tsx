import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Article from "../Article";
import { Router } from "react-router";
import { favoriteMock, unfavoriteMock, articleMock } from "./articleMockData";

import { createMemoryHistory } from "history";

describe("Articles Test Suite", () => {
    const history = createMemoryHistory();
    const wrapper = render(
        <Router history={history}>
            <Article favorite={favoriteMock} unfavorite={unfavoriteMock} article={articleMock} />
        </Router>
    );

    test("Article snapshot renders", () => {
        const previewJson = toJson(wrapper);
        expect(previewJson).toMatchSnapshot();
    });

    it("displays the mock article with the following data", () => {
        const articleAuthor = wrapper.find("#author");
        const articleCreatedAt = wrapper.find("#createdAt");
        const articleFavoritesCount = wrapper.find("#favoritesCount");
        const articleTitle = wrapper.find("#articleTitle");
        const articleDescription = wrapper.find("#articleDescription");

        console.log(articleAuthor.text(), articleCreatedAt.text(), articleFavoritesCount.text(), articleTitle.text(), articleDescription.text());

        expect(articleAuthor.text()).toEqual("john smith");
        expect(articleCreatedAt.text()).toEqual("Wed Jan 01 2020");
        expect(articleFavoritesCount.text()).toEqual(" 77");
        expect(articleTitle.text()).toEqual("this is a story");
        expect(articleDescription.text()).toEqual("mock description");
    });
});
