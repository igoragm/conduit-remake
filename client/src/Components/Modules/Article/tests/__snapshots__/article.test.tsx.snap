// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Test Suite Article snapshot renders 1`] = `
<div
  class="articlePreview"
  id="article-item"
>
  <div
    class="articleMeta"
  >
    <a
      href="/@john smith"
    >
      <img
        alt="john smith"
        class="articleMetaImg"
        src="mock image"
      />
    </a>
    <div
      class="info"
    >
      <a
        class="author"
        href="/@john smith"
        id="author"
      >
        john smith
      </a>
      <span
        class="date"
        id="createdAt"
      >
        Wed Jan 01 2020
      </span>
    </div>
    <div
      class="pullXsRight"
    >
      <button
        class="tagBtn"
        id="favoritesCount"
      >
        <i
          class="ionHeart"
        />
         77
      </button>
    </div>
  </div>
  <a
    class="previewLink"
    href="/article/mock slug"
  >
    <h1
      class="articlePreviewH1"
      id="articleTitle"
    >
      this is a story
    </h1>
    <p
      id="articleDescription"
    >
      mock description
    </p>
    <span>
      Read more...
    </span>
    <ul
      class="tagList"
      id="tags-list"
    >
      <li
        class="tag"
        id="articleTag_mock tagList"
      >
        mock tagList
      </li>
    </ul>
  </a>
</div>
`;
